### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I implemented the following:

- Created class Person and added required instance variables and methods.
- Created class Student, that inherited from Person class and added required variables and methods.
- Created class teacher, that inherited from Person class and added required variables and methods.
- Saved each class in its own file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+# student class
+class Student < Person
+  def initialize(classroom)
```
Comment (gemmen29) [inline student.rb]: - [x] Kindly pass the `age` and `name` to the initialize method so you can pass them to the super class.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+# teacher class
+class Teacher < Person
+  def initialize(specialization)
```
Comment (gemmen29) [inline teacher.rb]: - [x] Kindly pass the `age` and `name` to the initialize method so you can pass them to the super class.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Person class
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (gemmen29) [inline person.rb]: - [ ] Kindly change the `:` to `=` for `parent_permission` or change the way you call the super in order to make your code valid.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @EidHachem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (EidHachem) [reply person.rb]: Hello @gemmen29 I was using parent_permissin = true but it was giving me a linter error, and when I use parent_permission: true I can't change it to false when I instantiate a new person.
What is the other way to call the super so I can try it?

thank you so much

**Review by aimemalaika — APPROVED** [review]
Hi @EidHachem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (gemmen29) [reply person.rb]: Hi @EidHachem if you let `:` so you can call super like this:
```
super(age, name, parent_permission: parent_permission)
```

**Pull request merged**